CMakeLists.txt

```diff
@@ -22,6 +22,9 @@ set(CMAKE_CXX_STANDARD 17)
 # Disable C++ exceptions.
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions -fno-rtti")
 
+# Set `warning-as-error` for all the warnings except `comment` and `return-type`.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra -Wno-comment -Wno-return-type")
```

Comment:
`Werror` for all of these seems too aggressive to me. Depending on the compiler version, the project can fail to build on somebody else's system even if it passes our CI.

I'd enable the warnings here but move the `-Werror` to the cmake invocation in docker, so that warnings don't break local builds.